File: leet_GRDY_1282_regrouping.py
class Solution:
    def groupThePeople(self, groupSizes: List[int]) -> List[List[int]]:
        res = []
        D = {}
        for i in range(len(groupSizes)):
            L = groupSizes[i]
            if L not in D:
                D[L]  = []
            D[L].append(i)
            if len(D[L]) == L:
                res.append(D[L])
                D[L] = []
        return res

File: leet_GRDY_1282_regrouping.cc
class Solution {
public:
    vector<vector<int>> groupThePeople(vector<int>& groupSizes)
    {
        vector<vector<int>> res;
        unordered_map<int, vector<int>> D;
        int i = -1;
        while (++i < (int) groupSizes.size())
        {
            int len = groupSizes[i];
            D[len].push_back(i);
            if (D[len].size() == len)
            {
                res.push_back(D[len]);
                D[len].clear();
            }
        }
        return res;
    }
};
